bigquery/google/cloud/bigquery/magics.py

```diff
@@ -265,6 +276,14 @@ def _cell_magic(line, query):
     """
     args = magic_arguments.parse_argstring(_cell_magic, line)
 
+    if args.params is not None:
+        try:
+            params = json.loads(args.params)
+        except JSONDecodeError as e:
+            raise JSONDecodeError('--params is not a correctly formatted JSON string', e.doc, e.pos)
+
+        query = query.format(**params)
```

Comment:
I find this surprising behavior. I would expect this to use BigQuery's built-in query parameter handling. The DB-API implementation may be of good use here. See:

https://github.com/googleapis/google-cloud-python/blob/448e0ec97db246f3f42848ec8e92b946ed51ff4e/bigquery/google/cloud/bigquery/dbapi/cursor.py#L320

https://github.com/googleapis/google-cloud-python/blob/448e0ec97db246f3f42848ec8e92b946ed51ff4e/bigquery/google/cloud/bigquery/dbapi/_helpers.py#L96

and

https://github.com/googleapis/google-cloud-python/blob/448e0ec97db246f3f42848ec8e92b946ed51ff4e/bigquery/google/cloud/bigquery/dbapi/cursor.py#L148

Since DB-API format option is not required, I recommend skipping the `_format_operation` step.

Reply:
@tswast This reply was actually to your comment (easy to do with the Github UI being "helpful").

Reply:
The method

https://github.com/googleapis/google-cloud-python/blob/448e0ec97db246f3f42848ec8e92b946ed51ff4e/bigquery/google/cloud/bigquery/dbapi/_helpers.py#L96

does exactly what you want. It converts strings to the proper parameter objects.